osu.Game/Screens/Play/Player.cs

```diff
@@ -830,6 +830,24 @@ protected override bool OnScroll(ScrollEvent e)
 
         protected readonly Bindable<bool> LeaderboardExpandedState = new BindableBool();
 
+        private void loadSpectators()
+        {
+            var spectators = CreateGameplaySpectatorList();
+
+            if (spectators != null)
+            {
+                LoadComponentAsync(spectators, list =>
+                {
+                    if (!LoadedBeatmapSuccessfully)
+                        return;
+
+                    // TODO: On new spectate, add user
+                    // TODO: On spectate leave, remove user
```

Comment:
Are these supposed to be here?

Reply:
As far as i can tell, this functionality is still missing in the spectator-server. Intended to leave the TODOs here until that happens, or I'm being made aware of the way it already works, in case it exists and i cant find it.